app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeActivity.kt

```diff
@@ -304,9 +306,12 @@ class ComposeActivity : BaseActivity(),
                 setStatusVisibility(visibility)
             }
             viewModel.media.observe { media ->
-                composeMediaPreviewBar.visible(media.isNotEmpty())
                 mediaAdapter.submitList(media)
-                updateSensitiveMediaToggle(viewModel.markMediaAsSensitive.value != false, viewModel.showContentWarning.value != false)
+                if(media.size != mediaCount) {
```

Comment:
Any other idea how I could get the method to be only called when the count changes? Can you dedupe a `LiveData` like you can with Rx?

Reply:
Not in a built-in way

https://github.com/musichin/ReactiveLiveData/blob/master/reactivelivedata/src/main/java/com/github/musichin/reactivelivedata/ReactiveLiveData.kt#L317

https://github.com/adibfara/Lives/blob/master/lives/src/main/java/com/snakydesign/livedataextensions/Filtering.kt#L34

Reply:
I'm a little bit worried that it may change based on something else than media length and theoretically we should combine `media`. `markMediaAsSensitive` and `showContentWarning` into one thing but I think it's fine for now

Reply:
The only thing from `media` that is checked inside `updateSensitiveMediaToggle` is the length so it should be fine